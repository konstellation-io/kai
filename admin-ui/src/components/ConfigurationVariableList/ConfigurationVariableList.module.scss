@import '../../styles/colors';
@import '../../styles/mixins';

$color-input-bg: palette(base, 600);
$color-border: palette(base);
$bg-odd: palette(base);

.container {
  display: flex;
  flex-direction: column;
  margin: 50px 0 20px 0;
  height: 100%;
  overflow: hidden;

  .row:nth-child(odd),
  .row:nth-child(0) {
    &,
    .typeColValue {
      background-color: $bg-odd;
      color: font-color(light);
    }
  }
}

.row:last-child {
  border-bottom: solid 1px $color-border;
}

.row {
  @include font-body;
  line-height: 13px;
  font-weight: 600;

  display: flex;
  flex-direction: row;
  border-top: solid 1px $color-border;
  transition: color ease 0.3s, background-color ease 0.3s;

  &:hover {
    &,
    .typeColValue {
      background-color: $bg-color-dark !important;
      color: font-color(light);
    }

    input,
    textarea {
      color: font-color(light);
    }
  }

  > div:not(:last-child) {
    border-right: solid 1px $color-border;
  }

  > div {
    padding: 32px 20px 16px 30px;
  }

  .col1 {
    width: 20%;
    min-width: 140px;
  }
  .col2 {
    width: 30%;
    min-width: 170px;
  }
  .col3 {
    width: 50%;
    min-width: 240px;
  }

  .typeCol {
    display: flex;
    position: relative;
    padding: 16px 0;

    .typeColValue {
      padding: 0 8px 0 16px;
      background-color: $bg-color;
      transition: background-color ease 0.3s;
      width: fit-content;
      z-index: 2;
    }
  }

  .variableCol {
    display: flex;
    padding: 16px 0;

    > div {
      margin-left: 8px;
    }
  }

  .variableValue {
    > label {
      display: none;
    }

    > input,
    > textarea {
      background-color: $color-border;
    }

    > div {
      top: 13px;
    }
  }
}

.header {
  @include border-header;

  @include font-caption;
  font-weight: 800;

  min-height: 40px;
  line-height: 40px;
  border-top-width: 0;
  border-bottom: 2px solid #4f4f4f;

  > div {
    padding: 0 0 0 20px;
  }
}

.list {
  overflow: auto;
}
