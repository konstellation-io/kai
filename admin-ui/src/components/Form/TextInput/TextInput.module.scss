@import '../../../styles/mixins';
@import '../../../styles/colors';

.container {
  position: relative;

  &.white,
  &.white .input {
    color: white;
  }

  &.hasClearButton .input,
  &.hasEyeButton .input {
    padding-right: 30px;
  }

  &.hasClearButton.hasEyeButton .input {
    padding-right: 60px;
  }
}

.input {
  @include form-input;

  color: $font-color-regular;

  &::-webkit-input-placeholder {
    color: $font-color-regular;
  }

  &.lockHorizontalGrowth {
    resize: vertical;
  }
}

.clearButton {
  position: absolute;
  right: 15px;
  top: 34px;
  font-size: 15px;
  line-height: 11px;
  font-weight: 500;
  cursor: pointer;
  transition: color ease 0.3s;

  &:hover {
    color: $font-color-white;
  }
}

.eyeButton {
  position: absolute;
  right: 15px;
  top: 34px;
  cursor: pointer;
  transition: color ease 0.3s, right ease 0.3s;

  &:hover {
    color: $font-color-white;
  }

  &.showClearButton {
    right: 35px;
  }
}
