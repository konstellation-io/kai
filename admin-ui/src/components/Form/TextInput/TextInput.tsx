import React, {
  useState,
  useEffect,
  MouseEvent,
  KeyboardEvent,
  FocusEvent,
  ChangeEvent,
  ReactElement
} from 'react';

import InputLabel from '../InputLabel/InputLabel';
import InputError from '../InputError/InputError';
import InputHelp from '../InputHelp/InputHelp';
import IconShow from '@material-ui/icons/RemoveRedEye';
import IconHide from '@material-ui/icons/RemoveRedEyeOutlined';
import { isFieldAnInteger } from '../check';

import cx from 'classnames';
import styles from './TextInput.module.scss';

type Props = {
  onSubmit?: Function;
  onChange?: Function;
  onBlur?: Function;
  placeholder?: string;
  label?: string;
  textArea?: boolean;
  height?: number;
  lockHorizontalGrowth?: boolean;
  limits?: {
    minWidth?: number;
    maxWidth?: number;
    minHeight?: number;
    maxHeight?: number;
  };
  error?: string;
  helpText?: string;
  showClearButton?: boolean;
  whiteColor?: boolean;
  onlyNumbers?: boolean;
  positive?: boolean;
  formValue?: string | number;
  customClassname?: string;
  hidden?: boolean;
  autoFocus?: boolean;
  maxLength?: number;
};

function TextInput({
  onSubmit = function() {},
  onChange = function() {},
  onBlur = function() {},
  placeholder = '',
  label = '',
  textArea = false,
  height = 40,
  lockHorizontalGrowth = false,
  limits = {},
  error = '',
  helpText = '',
  showClearButton = false,
  whiteColor = false,
  onlyNumbers = false,
  positive = false,
  formValue = '',
  customClassname = '',
  hidden = false,
  autoFocus = false,
  maxLength
}: Props) {
  const [value, setValue] = useState(formValue);
  const [isHidden, setIsHidden] = useState(hidden);

  useEffect(() => {
    setIsHidden(hidden);
  }, [hidden, setIsHidden]);

  useEffect(() => {
    setValue(formValue);
  }, [formValue, setValue]);

  function updateValue(newValue: string) {
    if (
      !onlyNumbers ||
      (onlyNumbers && isFieldAnInteger(newValue, positive).valid)
    ) {
      setValue(newValue);
      onChange(newValue);
    }
  }

  function onType(e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
    const newValue = e.target.value;

    updateValue(newValue);
  }

  function onKeyPress(
    e: KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>
  ) {
    if (e.which === 13 && !textArea) {
      // Enter key
      onSubmit(value);
    }
  }

  function onInputBlur(e: FocusEvent<HTMLInputElement | HTMLTextAreaElement>) {
    onBlur();
  }

  function toggleVisibility(e: MouseEvent<HTMLDivElement>) {
    setIsHidden(!isHidden);
  }

  const inputProps = {
    className: cx(styles.input, {
      [styles.error]: error !== '',
      [styles.lockHorizontalGrowth]: lockHorizontalGrowth
    }),
    value: value,
    type: isHidden ? 'password' : 'text',
    placeholder: placeholder,
    onChange: onType,
    onKeyPress: onKeyPress,
    onBlur: onInputBlur,
    style: { height },
    maxLength: maxLength
  };
  const inputElement: ReactElement =
    textArea && !isHidden ? (
      <textarea
        {...inputProps}
        data-testid="input"
        style={{ ...limits }}
        autoFocus={autoFocus}
      />
    ) : (
      <input {...inputProps} data-testid="input" autoFocus={autoFocus} />
    );
  const cleanButton =
    showClearButton && value !== '' ? (
      <div
        className={styles.clearButton}
        onClick={() => updateValue('')}
        data-testid="clear-button"
      >
        x
      </div>
    ) : (
      ''
    );
  const VisibilityIcon = isHidden ? IconShow : IconHide;
  const showEyeButton = hidden ? (
    <div
      className={cx(styles.eyeButton, {
        [styles.showClearButton]: showClearButton && value !== ''
      })}
      onClick={toggleVisibility}
    >
      <VisibilityIcon className="icon-small" />
    </div>
  ) : (
    ''
  );

  return (
    <div
      className={cx(styles.container, customClassname, {
        [styles.white]: whiteColor,
        [styles.hasClearButton]: showClearButton,
        [styles.hasEyeButton]: hidden
      })}
    >
      <InputLabel text={label} />
      {inputElement}
      {cleanButton}
      {showEyeButton}
      {error ? (
        <InputError message={error} />
      ) : (
        <InputHelp message={helpText} />
      )}
    </div>
  );
}

export default TextInput;
