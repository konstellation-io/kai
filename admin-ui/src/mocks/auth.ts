import { GET_USER_EMAIL } from '../components/Header/Header.graphql';

export const usernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'user@konstellation.com'
      }
    }
  }
};

export const unauthorizedUsernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'unauthorizedUser@konstellation.com'
      }
    }
  },
  error: new Error('Unauthorized user')
};
