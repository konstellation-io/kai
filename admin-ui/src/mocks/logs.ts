import { GET_LOGS } from '../graphql/client/queries/getLogs.graphql';

export const currentLogPanelMock = {
  request: {
    query: GET_LOGS
  },
  result: {
    data: {
      runtimeId: 'runtimeIdMock',
      nodeId: 'nodeIdMock',
      nodeName: 'nodeNameMock'
    }
  }
};
