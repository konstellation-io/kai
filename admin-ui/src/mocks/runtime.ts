import { GET_DASHBOARD } from '../pages/Dashboard/Dashboard.graphql';
import { ADD_RUNTIME } from '../pages/AddRuntime/AddRuntime.graphql';

export const dashboardMock = {
  request: {
    query: GET_DASHBOARD
  },
  result: {
    data: {
      dashboard: {
        runtimes: [
          {
            id: '00001',
            status: 'ACTIVE',
            name: 'Some Name',
            creationDate: '2018-01-02'
          },
          {
            id: '00002',
            status: 'ACTIVE',
            name: 'Some Other Name',
            creationDate: '2018-02-03'
          }
        ],
        alerts: [
          {
            id: 'id0',
            type: 'error',
            message: 'some message',
            runtime: { id: '00001' }
          },
          {
            id: 'id1',
            type: 'error',
            message: 'some message 2',
            runtime: { id: '00001' }
          }
        ]
      }
    }
  }
};

export const addRuntimeMock = {
  request: {
    query: ADD_RUNTIME,
    variables: { name: 'New Runtime' }
  },
  result: {
    data: {
      createRuntime: {
        success: true,
        message: 'some message'
      }
    }
  }
};
