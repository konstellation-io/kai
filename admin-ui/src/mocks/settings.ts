import {
  GET_DOMAINS,
  GET_EXPIRATION_TIME,
  ADD_ALLOWED_DOMAIN,
  UPDATE_EXPIRATION_TIME
} from '../pages/Settings/Settings.graphql';
import { GET_USERS_ACTIVITY } from '../components/UserActivityList/UserActivityList.graphql';

export const domainMock = {
  request: {
    query: GET_DOMAINS
  },
  result: {
    data: {
      settings: {
        authAllowedDomains: ['domain.1', 'domain.2', 'domain.3.sample']
      }
    }
  }
};

export const usersActivityMock = {
  request: {
    query: GET_USERS_ACTIVITY
  },
  result: {
    data: {
      usersActivity: [
        {
          user: 'user1@domain.com',
          message: 'some message 1',
          date: '2019-01-01'
        },
        {
          user: 'user2@domain.com',
          message: 'some message 2',
          date: '2019-01-02'
        },
        {
          user: 'user3@domain.com',
          message: 'some message 3',
          date: '2019-01-03'
        }
      ]
    }
  }
};

export const expirationTimeMock = {
  request: {
    query: GET_EXPIRATION_TIME
  },
  result: {
    data: {
      settings: {
        cookieExpirationTime: 45
      }
    }
  }
};

export const addAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'intelygenz.com' }
  },
  result: {
    data: {
      addAllowedDomain: {
        authAllowedDomains: [
          'domain.1',
          'domain.2',
          'domain.3.sample',
          'intelygenz.com'
        ]
      }
    }
  }
};

export const removeAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'domain.3.sample' }
  },
  result: {
    data: {
      removeAllowedDomain: {
        authAllowedDomains: ['domain.1', 'domain.2']
      }
    }
  }
};

export const updateExpirationTime = {
  request: {
    query: UPDATE_EXPIRATION_TIME,
    variables: { input: { cookieExpirationTime: 10 } }
  },
  result: {
    data: {
      setSettings: {
        cookieExpirationTime: 12
      }
    }
  }
};
