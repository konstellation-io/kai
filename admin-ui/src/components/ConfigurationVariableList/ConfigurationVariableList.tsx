import { sortBy, capitalize } from 'lodash';

import React from 'react';

import TextInput from '../../components/Form/TextInput/TextInput';
import IconKey from '@material-ui/icons/VpnKey';
import TypeIcon from '@material-ui/icons/Code';

import { GetConfigurationVariables_version_configurationVariables as ConfVar } from '../../graphql/queries/types/GetConfigurationVariables';
import { ConfigurationVariableType } from '../../graphql/types/globalTypes';

import cx from 'classnames';
import styles from './ConfigurationVariableList.module.scss';

interface VariableRowProps {
  variable: ConfVar;
  onType: Function;
  hide: boolean;
}

export function VariableRow({
  variable: { type, key, value },
  onType,
  hide
}: VariableRowProps) {
  function onValueUpdate(inputValue: string) {
    onType(key, inputValue);
  }

  return (
    <div className={styles.row}>
      <div className={styles.col1}>
        <div className={styles.typeCol}>
          <TypeIcon className="icon-small" />
          <div className={styles.typeColValue}>{capitalize(type)}</div>
        </div>
      </div>
      <div className={styles.col2}>
        <div className={styles.variableCol}>
          <IconKey className="icon-small" />
          <div>{key}</div>
        </div>
      </div>
      <div className={styles.col3}>
        <TextInput
          label=""
          error={''}
          onChange={onValueUpdate}
          onSubmit={() => {}}
          customClassname={styles.variableValue}
          formValue={value}
          textArea={type === ConfigurationVariableType.FILE}
          limits={{
            minHeight: 95,
            maxHeight: 350
          }}
          showClearButton
          hidden={hide}
          lockHorizontalGrowth
        />
      </div>
    </div>
  );
}

type Props = {
  data: ConfVar[];
  onType: Function;
  hideAll: boolean;
};

function ConfigurationVariableList({ data, onType, hideAll }: Props) {
  const sortedData = sortBy(data, ['key']);
  const variableRows = sortedData.map((variable: ConfVar) => (
    <VariableRow
      onType={onType}
      key={variable.key}
      hide={hideAll}
      variable={variable}
    />
  ));

  return (
    <div className={styles.container}>
      <div className={cx(styles.header, styles.row)}>
        <div className={styles.col1}>TYPE</div>
        <div className={styles.col2}>KEY</div>
        <div className={styles.col3}>VALUE</div>
      </div>
      <div className={styles.list}>{variableRows}</div>
    </div>
  );
}

export default ConfigurationVariableList;
