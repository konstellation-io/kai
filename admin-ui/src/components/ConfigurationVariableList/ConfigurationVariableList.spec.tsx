import React from 'react';
import ConfigurationVariableList, {
  VariableRow
} from './ConfigurationVariableList';
import { confVarsMock } from '../../mocks/version';
import TextInput from '../Form/TextInput/TextInput';
import { shallow } from 'enzyme';

const DATA = confVarsMock.result.data.version.configurationVariables;
const VARIABLE_DATA = DATA[0];

let wrapper: any;
const mockOnType = jest.fn();

afterEach(() => {
  mockOnType.mockClear();
});

describe('ConfigurationVariableList', () => {
  beforeEach(() => {
    wrapper = shallow(
      <ConfigurationVariableList
        data={DATA}
        hideAll={true}
        onType={mockOnType}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.find(VariableRow).length).toBe(3);

    // varaibles should be sorted by default
    expect(
      wrapper
        .find(VariableRow)
        .at(0)
        .prop('variable').key
    ).toBe('var1');
  });

  it('calls onType properly', () => {
    wrapper
      .find(VariableRow)
      .at(1)
      .dive()
      .find(TextInput)
      .dive()
      .find('input')
      .simulate('change', { target: { value: 'new-value' } });

    expect(mockOnType).toHaveBeenCalledTimes(1);
  });

  test('all inputs are hidden by default', () => {
    for (let idx = 0; idx <= 2; idx++) {
      expect(
        wrapper
          .find(VariableRow)
          .at(idx)
          .prop('hide')
      ).toBeTruthy();
    }
  });

  test('all inputs all shown with hideAll prop set to false', () => {
    wrapper.setProps({ hideAll: false });

    for (let idx = 0; idx <= 2; idx++) {
      expect(
        wrapper
          .find(VariableRow)
          .at(idx)
          .prop('hide')
      ).toBeFalsy();
    }
  });
});

describe('VariableRow', () => {
  beforeEach(() => {
    wrapper = shallow(
      <VariableRow variable={VARIABLE_DATA} onType={mockOnType} hide={true} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.exists(TextInput)).toBeTruthy();
    expect(wrapper.find('.typeColValue').text()).toBe('Variable');
    expect(wrapper.find('.variableCol div').text()).toBe('var2');
  });

  it('calls onType properly', () => {
    wrapper
      .find(TextInput)
      .dive()
      .find('input')
      .simulate('change', { target: { value: 'new-value' } });

    expect(mockOnType).toHaveBeenCalledTimes(1);
  });
});
