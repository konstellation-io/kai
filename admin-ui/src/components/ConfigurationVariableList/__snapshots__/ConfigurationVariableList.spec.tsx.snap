// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationVariableList matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header row"
  >
    <div
      className="col1"
    >
      TYPE
    </div>
    <div
      className="col2"
    >
      KEY
    </div>
    <div
      className="col3"
    >
      VALUE
    </div>
  </div>
  <div
    className="list"
  >
    <VariableRow
      hide={true}
      key="var1"
      onType={[MockFunction]}
      variable={
        Object {
          "key": "var1",
          "type": "VARIABLE",
          "value": "value2",
        }
      }
    />
    <VariableRow
      hide={true}
      key="var2"
      onType={[MockFunction]}
      variable={
        Object {
          "key": "var2",
          "type": "VARIABLE",
          "value": "value1",
        }
      }
    />
    <VariableRow
      hide={true}
      key="var3"
      onType={[MockFunction]}
      variable={
        Object {
          "key": "var3",
          "type": "FILE",
          "value": "value3",
        }
      }
    />
  </div>
</div>
`;

exports[`VariableRow matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col1"
  >
    <div
      className="typeCol"
    >
      <CodeIcon
        className="icon-small"
      />
      <div
        className="typeColValue"
      >
        Variable
      </div>
    </div>
  </div>
  <div
    className="col2"
  >
    <div
      className="variableCol"
    >
      <VpnKeyIcon
        className="icon-small"
      />
      <div>
        var2
      </div>
    </div>
  </div>
  <div
    className="col3"
  >
    <TextInput
      customClassname="variableValue"
      formValue="value1"
      hidden={true}
      limits={
        Object {
          "maxHeight": 350,
          "minHeight": 95,
        }
      }
      lockHorizontalGrowth={true}
      onChange={[Function]}
      showClearButton={true}
      textArea={false}
    />
  </div>
</div>
`;
